source/common/secret/sds_api.h

```diff
@@ -276,5 +278,56 @@ class TlsSessionTicketKeysSdsApi : public SdsApi, public TlsSessionTicketKeysCon
       validation_callback_manager_;
 };
 
+/**
+ * GenericSecretSdsApi implementation maintains and updates dynamic generic secret.
+ */
+class GenericSecretSdsApi : public SdsApi, public GenericSecretConfigProvider {
+public:
+  static GenericSecretSdsApiSharedPtr
+  create(Server::Configuration::TransportSocketFactoryContext& secret_provider_context,
+         const envoy::config::core::v3::ConfigSource& sds_config,
+         const std::string& sds_config_name, std::function<void()> destructor_cb) {
+    // We need to do this early as we invoke the subscription factory during initialization, which
+    // is too late to throw.
+    Config::Utility::checkLocalInfo("GenericSecretSdsApi", secret_provider_context.localInfo());
+    return std::make_shared<GenericSecretSdsApi>(
+        sds_config, sds_config_name, secret_provider_context.clusterManager().subscriptionFactory(),
+        secret_provider_context.dispatcher().timeSource(),
+        secret_provider_context.messageValidationVisitor(), secret_provider_context.stats(),
+        *secret_provider_context.initManager(), destructor_cb);
+  }
+
+  GenericSecretSdsApi(const envoy::config::core::v3::ConfigSource& sds_config,
+                      const std::string& sds_config_name,
+                      Config::SubscriptionFactory& subscription_factory, TimeSource& time_source,
+                      ProtobufMessage::ValidationVisitor& validation_visitor, Stats::Store& stats,
+                      Init::Manager& init_manager, std::function<void()> destructor_cb)
+      : SdsApi(sds_config, sds_config_name, subscription_factory, time_source, validation_visitor,
+               stats, init_manager, std::move(destructor_cb)) {}
+
+  // SecretProvider
+  const envoy::extensions::transport_sockets::tls::v3::GenericSecret* secret() const override {
+    return generic_secret.get();
+  }
+  Common::CallbackHandle* addValidationCallback(
+      std::function<void(const envoy::extensions::transport_sockets::tls::v3::GenericSecret&)>)
+      override {
+    return nullptr;
```

Comment:
`NOT_IMPLEMENTED_GCOVR_EXCL_LINE` instead? Why this isn't implemented btw?

Reply:
I am just following how it is "implemented" for other secret types. I think it worth a separate PR to consistently implement the method for all secret types. That being said, do you want me to replace `return nullptr` with `NOT_IMPLEMENTED_GCOVR_EXCL_LINE` in all `XxxSdsApi` classes?

Reply:
`return nullptr` is only for the case of `TlsCertificate` because we didn't expect additional verification (more than proto validation) to be done on that, but we do implement in SessionTicketKeys and ValidationContext: https://github.com/envoyproxy/envoy/blob/master/source/common/secret/sds_api.h#L186-L191

Reply:
My apology for missing that. I have pushed the implementation and unit test.